search/search-client-elasticsearch/src/main/java/io/camunda/search/es/transformers/aggregator/SearchFiltersAggregatorTransformer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.es.transformers.aggregator;
+
+import co.elastic.clients.elasticsearch._types.aggregations.Aggregation;
+import co.elastic.clients.elasticsearch._types.aggregations.Buckets;
+import co.elastic.clients.elasticsearch._types.aggregations.FiltersAggregation;
+import co.elastic.clients.elasticsearch._types.query_dsl.Query;
+import io.camunda.search.clients.aggregator.SearchFiltersAggregator;
+import io.camunda.search.es.transformers.ElasticsearchTransformers;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+public class SearchFiltersAggregatorTransformer
+    extends AggregatorTransformer<SearchFiltersAggregator, Aggregation> {
+
+  public SearchFiltersAggregatorTransformer(final ElasticsearchTransformers transformers) {
+    super(transformers);
+  }
+
+  @Override
+  public Aggregation apply(final SearchFiltersAggregator value) {
+    final var queries =
```

Comment:
❓ Should we add a null safety point here? Or this state never will happen?

Reply:
If users use the builder from the record or from the `SearchAggregatorBuilders` then this can't happen. So I don't think it's super necessary.
The record itself could allow `null` queries.